pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -216,20 +244,44 @@ public void setPassword(String password) {
    * Gets the port which the PostgreSQL server is listening on for TCP/IP connections.
    *
    * @return The port, or 0 if the default port will be used.
+   * @deprecated use {@link #getPortNumbers()}
    */
+  @Deprecated
   public int getPortNumber() {
-    return portNumber;
+    return portNumbers[0];
```

Comment:
This assumes `portNumbers` is not null but other parts of the code do that check prior to use.

Reply:
Hmmm I'd still like to be a little more conservative here and check it anyway. If someone changes the code down the road and removes the other check

Reply:
done